### PR TITLE
[FIX] find confounds even when BIDS derivatives bold files contain res or den entities

### LINKED ISSUE #3741: [BUG] first_level_from_bids confused by "res" filter when looking for confounds.tsv
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Operating system

- [X] Linux
- [ ] Mac
- [ ] Windows

### Operating system version

CentOS Linux release 7.6.1810

### Python version

- [ ] 3.11
- [X] 3.10
- [ ] 3.9
- [ ] 3.8
- [ ] 3.7

### nilearn version

0.10.0

### Expected behavior

When running the command below one can expect the model_confounds to be found in a bids-valid directory.

(models,
    models_run_imgs,
    models_events,
    models_confounds,) = first_level_from_bids(data_dir, task_label, derivatives_folder=derivatives_folder,space_label=space_label,img_filters=[("desc", "preproc"),("res","2")], smoothing_fwhm=6, slice_time_ref=0.481, n_jobs=-1, signal_scaling=(0,1))

### Current behavior & error messages

This is what I got:
models_confounds were al None.

What I did to fix it was simply add an extra line in /nilearn/glm/first_level/first_level.py for removing "res" from the filters:

    supported_filters.remove("desc")
    **supported_filters.remove("res")**
    
    filters = _make_bids_files_filter(
        task_label=task_label,
        supported_filters=supported_filters,
        extra_filter=img_filters,
        verbose=verbose,
    )
    confounds = get_bids_files(
        derivatives_path,
        modality_folder="func",
        file_tag="desc-confounds*",
        file_type="tsv",
        sub_label=sub_label,
        filters=filters,
    )

### Steps and code to reproduce bug

```python
# Paste your code here


```

### PR DESCRIPTION
**Pull request by Remi-Gau**
Fixes #3741

Changes proposed in this pull request:

- make test for `first_level_from_bids` more stringent by checking the types of the output values
- fix bug by not allowing any derivative entities when filtering for confounds

### CODE REVIEW

**Review by bthirion — APPROVED** [review]
LGTM, thx

File: nilearn/glm/tests/test_first_level.py [hunk 1]
```diff
@@ -1216,13 +1219,36 @@ def test_first_level_from_bids_several_labels_per_entity(tmp_path, entity):
...
+    check_output_first_level_from_bids(n_sub,
+                                       models,
+                                       m_imgs,
+                                       m_events,
+                                       m_confounds)
+    n_imgs_expected = n_ses * n_runs[0]
+    assert len(m_imgs[0]) == n_imgs_expected
+
+
+def check_output_first_level_from_bids(n_sub,
```
[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/glm/tests/test_first_level.py:1232]: Well it is in a test folders which I assume would not be public. Let me check what other utility functions are called in test folders.

[the remark this answers is not in the document]
Comment (ymzayek) [reply nilearn/glm/tests/test_first_level.py:1232]: True so maybe it's overkill. I have seen both. The idea is whether or not it's useful for the test function names to be telling if they are only used in that script or outside of it. But it doesn't really matter until we set some guidelines (#3613) and enforce them so we could leave it for now

[the remark this answers is not in the document]
Comment (Remi-Gau) [reply nilearn/glm/tests/test_first_level.py:1232]: actually better be explicit about things so I will add make the function explicitly private

**Pull request merged**